kbase-extension/static/kbase/js/narrativeConfig.js

```diff
@@ -105,62 +105,59 @@ define([
-                config.exampleData = dataCategories[config.environment].exampleData;
-                return Promise.try(function () {
-                    return config;
                 });
-            })
-            .catch(function (error) {
-                console.error('Config: unable to process remote data configuration options. Searching locally.');
-                // hate embedding this stuff, but it seems the only good way.
-                // the filename is the last step of that url path (after the last /)
-                var path = config.urls.data_panel_sources.split('/');
+            } else {
+                resolve(config);
+            }
+        }).then(function (config) {
+            console.log('Config: fetching remote data configuration.');
+            return Promise.resolve($.getJSON(config.urls.data_panel_sources));
+        }).then(function (dataCategories) {
+            console.log('Config: processing remote data configuration.');
+            config.publicCategories = dataCategories[config.environment].publicData;
+            config.exampleData = dataCategories[config.environment].exampleData;
+            return Promise.try(function () {
+                return config;
+            });
+        }).catch(function (error) {
+            console.error('Config: unable to process remote data configuration options. Searching locally.');
+            // hate embedding this stuff, but it seems the only good way.
+            // the filename is the last step of that url path (after the last /)
+            var path = config.urls.data_panel_sources.split('/');
 
-                return Promise.resolve($.getJSON('static/kbase/config/' + path[path.length - 1]))
-                    .then(function (dataCategories) {
-                        console.log('Config: processing local data configuration.');
-                        config.publicCategories = dataCategories[config.environment].publicData;
-                        config.exampleData = dataCategories[config.environment].exampleData;
-                        return Promise.try(function () {
-                            return config;
-                        });
-                    })
-                    .catch(function (error) {
-                        console.error('Config: unable to process local configuration options, too! Public and Example data unavailable!');
-                        return Promise.try(function () {
-                            return config;
-                        });
+            return Promise.resolve($.getJSON('static/kbase/config/' + path[path.length - 1]))
+                .then(function (dataCategories) {
+                    console.log('Config: processing local data configuration.');
+                    config.publicCategories = dataCategories[config.environment].publicData;
+                    config.exampleData = dataCategories[config.environment].exampleData;
+                    return Promise.try(function () {
```

Comment:
and these Promse.trys are not necessary since they simply return a value -- just returning the value will continue the promise chain.